### DISCUSSION TITLE
Welcome to esp32_8ch_motor_shield Discussions!

### DISCUSSION THREAD

**Discussion by nliaudat** [nliaudat/esp32_8ch_motor_shield#13] [Announcements]
## 👋 Welcome!
  We’re using Discussions as a place to connect with other members of our community. We hope that you:
  * Ask questions you’re wondering about.
  * Share ideas.
  * Engage with other community members.
  * Welcome others and are open-minded. Remember that this is a community we
  build together 💪.

  To get started, comment below with an introduction of yourself and tell us about what you do with this community.

**Comment by linuxdep** [comment]
Nice that you open discussion option!

**Comment by BartVB** [comment]
Just found this repo by coincidence. Just fantastic!
I've bought a number of HmIP-VDMOT motors, several HmIPW-FALMOT-C12 units and a HmIPW-DRAP to control our underfloor heating. Normally I do my due diligence, but this time I was in a hurry and it all looked like a very solid product, what could go wrong?

Everything worked as I expected, except for the fact that I can't control the valves directly 😞 I had expected the system to have a CCU API to tell a valve to open 70%, that's all it needed to do. But instead you need to use a Homematic Thermostat which will talk with the FALMOTs without giving the HA system any control over what happens. I don't want Homematic Thermostats, I already have KNX for that.

So this project is exactly what I need! I really like the proportional and energy efficient nature of the VDMOT motors, but need an alternative means to control them. This project is exactly that!

So if anyone is interested in buying several FALMOTs and a DRAP unit; they are taking up shelve space over here :)

Thank you, @nliaudat for sharing this project!

**Reply by nliaudat** [reply]
Many thanks !
If you are in early stage, you may find interesting to have choice between other alternative projects : 

- https://github.com/Lenti84/VdMot_Controller
- https://github.com/deklaus/OpenValveControl

**Reply by BartVB** [reply]
Wow, thanks!
VdMot_Controller might even be a better fit; I need more than 8 loops on three out of four floors.

I'll investigate the options when I have a bit more time. Thanks again!

**Reply by nliaudat** [reply]
Mine is cheaper :) 
You can have 2 boards (16 channels) for the price of one VdMot_Controller. 
But the 1st thing to have in mind is how you want to control your system ?
My firmware use esphome and is very linked with hass.io. It can work standalone with web interface activated
But my hardware cannot be bought ready to go and you have to place an order at jlcpcb (or other) or you can buy one in the community (in the discussion board)

**Comment by Karl-opec** [comment]
After a while i have looked again in this great project. I  tried the v61 board and have some questions / ideas to it.

The Motor H-Driver are now TC118S, which is great, but now a 1N4007 Diode is used as a Shunt. What is the Reason for this? Vf die forward voltage is now approx. 0.67 V this reduces the total Voltage over the motor by nearly 20%  to 2.63 V instead of the 3.3V nominal to the motor. The Torque of the Motor is therefore dramaticly reduced. Sometime it even doen't make it to the endstop. I would propose to change this diode to MBR0530T1G, LCSC #C7436545 Price approx 1cent und  plug In Subst. this has a Vf of about 0.35 V and therefore the motor gets close to 3V. Torque is much better.

Why are 8 Diodes used the Diodes of CH-1-2 Ch 3-4 CH-5-6 CH-7-8 are switch in parallel. So 4 Diode could be eliminated.

**Reply by nliaudat** [reply]
Hi, 
Thank you for your proposal.
It took me a while to realise that, without a diode between GND and BEMF, the TC118S chip cannot provide a BEMF voltage reading.
That's why there are so many diodes.
The explanation for one diode per channel is simply a design simplification (I found no easy way to link two channels with one diode).

The choice of 1N4007 was simple: it's a very common component with high stock availability at JLCPCB.
JLCPCB has the bad habit of not refurbishing some parts, which is why I had to change the old L9110S chip, which was great but no longer available. (I needed to pre-order the component from LCSC.)

I've never had a torque problem with Homematic valves as they have a huge gearbox. Maybe your BEMF endstop limit is not enough?

In fact, the card is designed to supply 5V to the motors. This is possible because manufacturers do not make different coils for 1-6V motors. The coils are different for 6-12V motors. 
You can therefore normally power your motors with 5V without any problems. (Not tested with Homematic). 
I do this with my own valves because they use a very small 12gan20 type motor.

Another way to fix the issue is to lower the pressure in your hydraulic circuit if possible. 

Regards

**Reply by rzerres** [reply]
Hi,

Since i have to choose my valves I'd like to ask, if the valves you use with the 12gan20 type motor are available from some stock?
Is this you [`floor-heating-proportional-valve`](https://github.com/nliaudat/floor-heating-proportional-valve) repro?

I do need 6 valves.

Regards Ralf

**Reply by nliaudat** [reply]
Hi, 
The [floor-heating-proportional-valve](https://github.com/nliaudat/floor-heating-proportional-valve) is my repo and valve needs to be 3 printed. 
Regards
